lib/seahorn/BvOpSem2.cc

```diff
@@ -765,11 +770,17 @@ class OpSemMemManager {
     Expr val = _val;
     Expr mem = m_ctx.read(memReadReg);
 
+    unsigned offsetBits = getByteAlignmentBits();
+    bool wordAligned = offsetBits == 0 || align % wordSzInBytes() == 0;
```

Comment:
are you sure `align % wordSzInBytes() == 0` is correct test?

Reply:
`align` is the known alignment of pointers (given pointers are divisible by `align`, they are also divisible by all factors of `align`), so I believe it should be aligned to the word size (pointers are divisible by word size) as long as `align` is divisible by the word size

Reply:
what I am worried about is what happens when `wordSzInBytes() > align`

Reply:
as long as align isn't zero, the check will evaluate to false, which is intended in that case since a larger word size than known alignment means that the address is not necessarily word aligned

Reply:
got it